includes/database/class-give-db-meta.php

```diff
@@ -150,7 +150,7 @@ public function get_meta( $id = 0, $meta_key = '', $single = false ) {
 
 		if ( $this->raw_result ) {
 			if ( ! ( $value = get_metadata( $this->meta_type, $id, $meta_key, false ) ) ) {
-				$value = '';
+				$value = $single ? '' : array();
 			}
```

Comment:
@ravinderk What's the logical difference between this and:
```php
$value = get_metadata( $this->meta_type, $id, $meta_key, $single );
$this->raw_result = false;
```
Looks to me like the whole if-statement is unnecessary. This ternary is just reproducing what `get_metadata()` already does.

Reply:
@JasonTheAdams There is a big difference. This part of the code only runs if the `get_post_metadata` filter hook triggers (FYI, this function can be called directly).  This means logic uses the `get_post_data` WordPress function to get GiveWP code post type details instead of the core function. At the filter hook, we should return raw data and let the function decide the final result, otherwise, the result will be affected.

![image](https://github.com/impress-org/givewp/assets/1784821/d74e05a9-d775-42b7-9b2b-89df1ecd4c4a)


If this minimum change affects anything, I am happy to look deeper.

I am looking forward to your reply on https://github.com/impress-org/givewp/pull/7062#issuecomment-1780423425

Reply:
Sorry, it still doesn't add up in my mind. Here's how it works as far as I can see:
1. If it wants a "raw result" then it uses `get_metadata` to retrieve the value, always as an array
    a. If it's `$single`, return an empty string, otherwise an empty array (This behavior is the same as passing `$single` to `get_metadata`)
    b. Reset the "raw result" flag
2. If it does not want a "raw result", use `get_metadata` and pass `$single` as-is

So the only difference I can see if that `raw_result` _is not_ reset, if it's false. _BUT_, since it's already false, it doesn't need to be reset. Therefore, it will always end the function as false no matter what.

Therefore, with the changes being made here, there's no ultimate difference in behavior than simply passing `$single` to `get_metadata` and always resetting `$this->raw_result`.

Am I missing something?

Reply:
Also, that filter usage in the screenshot makes no sense to me. Why would this ever be true?
```php
$single && is_array($check)
```
If what's stored is a _single_ serialized array, we wouldn't want to arbitrarily grab the first item. This is the kind of logic that's bizarre. I don't understand why we ever did that.

Reply:
@ravinderk and I hopped on a call and he explained that this behavior is specific to when this is used in the meta filter, in which case we need to return an array value, as WordPress handles extracting the single value.